include/utils/point_locator_base.h

```diff
@@ -103,6 +103,7 @@ class PointLocatorBase : public ReferenceCountedObject<PointLocatorBase>
    * the subdomains searched.
    */
   virtual const Elem * operator() (const Point & p,
+                                   bool use_cache = true,
```

Comment:
This definitely doesn't work here.  You can't add a new parameter in the *middle* of a function signature, even with a default value, without breaking API backwards compatibility for anyone using the old signature.

I like the option (especially being able to easily select it per-evaluation rather than making it a property of the locator), though; try just moving use_cache to be after allowed_subdomains instead of before?

We'll also need some kind of documentation in the comment above.  I think pointing out the TOLERANCE-scale cache bias at element boundaries is a necessity here; if you hadn't explained that in the PR I'd have never imagined why the cache would ever cause a problem or why a reasonable solution would be to avoid caching rather than fix the problem.

Reply:
Fair point. That was a little shortsighted. I'll move the new argument to the end of signature.

I'll look into updating the documentation as well.